src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,151 @@
+package duke;
+
+import duke.commands.ByeCommand;
+import duke.commands.Command;
+import duke.commands.DeadlineCommand;
+import duke.commands.DeleteCommand;
+import duke.commands.DoneCommand;
+import duke.commands.EventCommand;
+import duke.commands.ListCommand;
+import duke.commands.ToDoCommand;
+import duke.commands.FindCommand;
+import duke.tasks.Deadline;
+import duke.tasks.Event;
+import duke.tasks.ToDo;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+/**
+ * Main function to breakdown the input of the user
+ */
+public class Parser {
+
+    private static final String TIME_PARSE_ERROR = "Hmm.. Seems like the time format is foreign to me. \n"
+            + "Please use the following format:\n"
+            + "yyyy-MM-dd HH:MM (e.g 2020-05-19 15:30)";
+
+    /**
+     * Interprets user input and returns the appropriate command
+     *
+     * @param userInput The string which the user typed in.
+     * @return Appropriate command the input represents for the main function to execute.
+     * @throws DukeExceptions if the input is lacking argument.
+     * @throws DukeExceptions if the input has too much argument.
+     * @throws DukeExceptions if the input is not properly formatted.
+     * @throws DukeExceptions if the input is empty.
+     * @throws DukeExceptions if the first word of the input is not a recognisable command.
```

Comment:
Is this good practice to have so many lines of javadocs for DukeExceptions? 

Personally I feel you could break up the exceptions into separate classes extended from `DukeException`!